go.mod

```diff
@@ -274,3 +274,5 @@ require (
 	google.golang.org/protobuf v1.31.0
 	gopkg.in/yaml.v3 v3.0.1
 )
+
+replace github.com/vito/progrock => github.com/jedevc/progrock v0.0.0-20231227162308-507a0333689a
```

Comment:
Doing this hack just for a moment - @vito does attaching as a label to a vertex seem like the right approach to you? If so, I can raise a PR upstream to let labels be attached here, though I'm not 100% sure if/how we'd want to display these in the TUI.

Reply:
It's definitely a reasonable approach, since there's precedent (groups already have labels). There's significant enough overlap here with what I did for DagQL/v2 IDs though that it could be worth consolidating. Instead of adding labels to vertices, I added a new type of message (`VertexMeta`) that associates a name + arbitrary protobuf value (`pb.Any`) to a vertex:

https://github.com/vito/progrock/commit/ec90ac8b55c1174ef5295c3f573ec9c4feb8c3cb#diff-7f68afa99b4b67604c1528cbce81f87996c978e3f180cd23663846a711a53160R151

Metadata is sent in a separate call after creating the vertex:

https://github.com/vito/progrock/commit/ec90ac8b55c1174ef5295c3f573ec9c4feb8c3cb#diff-648d54a3f982f918dc185d16a00c7170a472355d0b73bd68b8216d3c087c617dR144-R155

The use case for this is to send the v2 ID payload along so the frontend can unpack and visualize it. v2 IDs are already protobuf and I didn't want to double-encode it, so I reached for `pb.Any`. This was all hastily added to get the data from point A to point B so I can spike on visualization work; not sure if we'll even be using this protocol when all is said and done. But we could consider doing the same here, since the data we're sending is already Protobuf, right?

As for why it's a separate message vs. just adding a list of metadata fields to a Vertex, honestly there's not a huge reason here. The only physical difference is that we only send the metadata once rather than twice (once when the vertex appears and again when it finishes). The trade-off is you have to think about the state when metadata for a vertex hasn't arrived yet. Not sending twice _could_ be helpful since IDs can still get pretty large, and there's precedent for decoupling messages/events this way. But again this was pretty hastily put together.

All that being said, adding labels to vertices in Progrock feels somewhat inevitable, so if we want to just stick with that anyway I'm still good with that.